x/virtualgroup/keeper/payment.go

```diff
@@ -48,42 +32,24 @@ func (k Keeper) SettleAndDistributeGVGFamily(ctx sdk.Context, spID uint32, famil
 
 func (k Keeper) SettleAndDistributeGVG(ctx sdk.Context, gvg *types.GlobalVirtualGroup) error {
 	paymentAddress := sdk.MustAccAddressFromHex(gvg.GetVirtualPaymentAddress())
-	streamRecord, found := k.paymentKeeper.GetStreamRecord(ctx, paymentAddress)
-	if !found {
-		return nil
-	}
-
-	totalBalance := streamRecord.StaticBalance
-	diff := ctx.BlockTime().Unix() - streamRecord.CrudTimestamp
-	totalBalance = totalBalance.Add(streamRecord.NetflowRate.MulRaw(diff))
-	if !totalBalance.IsPositive() {
-		return nil
-	}
-
-	change := paymenttypes.NewDefaultStreamRecordChangeWithAddr(paymentAddress).WithStaticBalanceChange(totalBalance.Neg())
-	err := k.paymentKeeper.UpdateStreamRecord(ctx, streamRecord, change)
+	totalBalance, err := k.paymentKeeper.QueryDynamicBalance(ctx, paymentAddress)
 	if err != nil {
-		return fmt.Errorf("fail to settle gvg: %d, err: %s", gvg.Id, err.Error())
+		return fmt.Errorf("fail to query balance: %s, err: %s", paymentAddress.String(), err.Error())
 	}
-	k.paymentKeeper.SetStreamRecord(ctx, streamRecord)
 
 	amount := totalBalance.QuoRaw(int64(len(gvg.SecondarySpIds)))
```

Comment:
Do we need to think about the case where it's not divisible?

Reply:
If there is dust, then it still in payment account, which can be withdraw later.
Meanwhile, there should be no dust - when calculating Flow Rate, there is a multiplier `len(gvg.SecondarySpIds)`.